src/components/Chat/ChatTranscriptor/ChatMessages/InteractiveMessage.js

```diff
@@ -151,35 +152,68 @@ InteractiveMessage.propTypes = {
 };
 
 export function InteractiveMessage({ content, templateType, addMessage, textInputRef, isCarouselElem, templateIdentifier }) {
-  const [ responseSelected, setResponseSelected] = useState(false);
+  const [responseSelected, setResponseSelected] = useState(false);
+  const ref = useRef();
 
-  function onAddMessage(data){
+  function onAddMessage(data) {
     addMessage(data);
     setResponseSelected(true);
   }
 
-  useLayoutEffect(()=> {
+  /**
+   * use useEffect to set the necessary attributes and listeners required to 
+   * render views and get response from view
+   */
+  useEffect(() => {
+    // function to create chat message from view event
+    const viewOnActionCallback = ((event) => {
+      const actionData = event.detail;
+
+      const reshapedMessage = {
+        action: actionData.Action,
```

Comment:
What if this is an empty string?

Reply:
View actions today are not allowed to be empty